### DISCUSSION TITLE
WEbP images for all media folders

### DISCUSSION THREAD

**Discussion by nkarthickannan** [breezefront/community#329] [1. Help]
@vovayatsyuk 

I followed this image optimisation article - https://breezefront.com/docs/next-gen-images using nGinx and it is working fine. But one thing I have noted is that this optimisation is working only for the catalog related images. However, apart from catalog images, we have images in other directories as well - media/wysiwyg, media/customer/ etc.

As this optimisation is working only for the catalog images, the Google Pagespeed is reducing the score and asked to optimise these images as well. Is there any reason why the documentation is suggesting only to optimise the catalog images but not others ? If there are no reason, can you please let me know how to achieve this for all images.

**Comment by vovayatsyuk** [comment]
There is no reason why we didn't do that for all images. You can try to modify the code to make it work with other images too and write back here.

**Reply by nkarthickannan** [reply]
@vovayatsyuk 

I changed the code as below and it works fine for me. Can you please check if all looks good ? If so, please update the documentation so that other users can use it.

`File name : magento_root/nginx.conf.sample`
```
location ~* ^/media/.* { ## This is the modified line
    proxy_pass http://$resize_stream;
    proxy_cache images;
    proxy_cache_valid 200 24h;
}
```

`File name : /etc/nginx/sites-enabled/magento.conf`

```
# server for modern browsers
server {
    server_name localhost;
    listen 22222;
    set $MAGE_ROOT /www/magento2;
    root $MAGE_ROOT/pub;

    location ~* ^/(?<path>media/.*)$ { ## This is the modified line
        set $ext "";

        # when request is made for image.jpg,
        # check if image.jpg.webp is available.
        if (-f $MAGE_ROOT/pub/$path.webp) {
            set $ext .webp;
        }

        alias $MAGE_ROOT/pub/$path$ext;

        set $width "-";
        set $height "-";
        if ($arg_width != '') {
            set $width $arg_width;
        }
        # comment the section below to resize by width only
        if ($arg_height != '') {
            set $height $arg_height;
        }
        image_filter resize $width $height;
        image_filter_interlace on;
        image_filter_jpeg_quality 75;
        image_filter_webp_quality 75;
    }
}
```

Finally, the command to convert the images:
```
find ./pub/media/ -type d -name cache -prune -o \( -type f -name '*.jpg' -o -type f -name '*.png' \) -print0 | xargs -0 -P $(nproc) -I {} sh -c 'test ! -f $1.webp && echo $1.webp && cwebp -quiet -q 80 $1 -o $1.webp' _ {} \;
```

**Reply by vovayatsyuk** [reply]
Thank you!

I've updated the docs: https://breezefront.com/docs/next-gen-images
